### PR TITLE
Version management rework

### PR DESCRIPTION
**Pull request by avalentino**
The version management has been reworked according to astropy/astropy#10774.
There are some differences due to the custom versioning schemed used by pyerfa, but the basic concept should be the same.

### CODE REVIEW

File: erfa/_dev/scm_version.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        if not erfa_version.exact:
+            warn(f'liberfa/erfa not at a tagged release, but at {erfa_version}')
+
+        erfa_tag = erfa_version.format_with("{tag}")
+        version_string = str(version.tag)
+
+        if version.exact:
+            if not version_string.startswith(erfa_tag):
+                warn(
+                    f'tag {version_string} does not start with '
```
Comment (mhvk) [inline erfa/_dev/scm_version.py:28]: Nit: put on the same line as `warn(`

File: erfa/_dev/scm_version.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            else:
+                guessed = version_string.partition("+")[0] + '.1'
+            return version.format_with("{guessed}.dev{distance}",
+                                       guessed=guessed)
+
+    get_version = functools.partial(_get_version,
+                                    root=pth.join('..', '..'),
+                                    version_scheme=_guess_next_dev,
+                                    relative_to=__file__)
+except Exception:
```
Comment (mhvk) [inline erfa/_dev/scm_version.py:47]: I realize this comes straight from astropy, but I wonder if it isn't better to give at least a clue about what went wrong, by doing `except Exception as exc` and then `raise ImportError(...) from exc`

File: setup.py [hunk 3]
```diff
@@ -140,85 +138,27 @@ def get_extensions():
...
+except FileNotFoundError:
+    guess_next_dev = None
 else:
-    del get_version, _guess_next_dev
-""".lstrip()
-
-
-guess_next_dev = functools.partial(_guess_next_dev, liberfadir=LIBERFADIR)
-
+    import types
```
Comment (mhvk) [inline setup.py:147]: Is it still unavoidable to read and execute the file? If our tags are set up right, shouldn't it just work like astropy?

**Review by mhvk — COMMENTED** [review]
Some total nitpicks, but mostly a larger question, whether it is really unavoidable to read the `_dev/scm_version` file - it seems to slightly beat the point...

**Comment by mhvk** [comment]
p.s. Good to change all those remaining astropy references!

File: erfa/_dev/scm_version.py (hunk 1 above)
Comment (avalentino) [reply erfa/_dev/scm_version.py:28]: +1

File: erfa/_dev/scm_version.py (hunk 2 above)
Comment (avalentino) [reply erfa/_dev/scm_version.py:47]: yes, I perfectly agree

File: setup.py (hunk 3 above)
Comment (avalentino) [reply setup.py:147]: Well, if I make a source distribution in my working branch I get:  ``pyerfa-1.7.1.1.dev15+g3333cb6.tar.gz``.
If I remove that part that loads ``get_next_dev`` I get ``pyerfa-1.7.2.dev16+g2f17d0f.tar.gz`` instead.

**Comment by avalentino** [comment]
> Some total nitpicks, but mostly a larger question, whether it is really unavoidable to read the _dev/scm_version file - it seems to slightly beat the point...

IMHO if we want to keep out versioning schema then we need it. I mean I can't figure out a smarted way to achieve the same result.
I agree with you that it intoduces some complication ...

**Comment by avalentino** [comment]
OK, merging. We can eventually address remaining open points in a new PR maybe.

**Pull request merged**